From 82fa625540abf8b8ec23d43c41e2ca906a9928a5 Mon Sep 17 00:00:00 2001
From: Andreas Schneider <asn@samba.org>
Date: Fri, 15 Apr 2016 11:56:08 +0200
Subject: [PATCH] s3:rpc_server: Fix a regression verifying the security
 trailer

We do not support header signing so we should not check verify it if a
client sends the flag.

Signed-off-by: Andreas Schneider <asn@samba.org>
Reviewed-by: Guenther Deschner <gd@samba.org>
---
 source3/rpc_server/srv_pipe.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/source3/rpc_server/srv_pipe.c b/source3/rpc_server/srv_pipe.c
index d659705..fa354a6 100644
--- a/source3/rpc_server/srv_pipe.c
+++ b/source3/rpc_server/srv_pipe.c
@@ -1552,7 +1552,6 @@ static bool srv_pipe_check_verification_trailer(struct pipes_struct *p,
 {
 	TALLOC_CTX *frame = talloc_stackframe();
 	struct dcerpc_sec_verification_trailer *vt = NULL;
-	const uint32_t bitmask1 = 0;
 	const struct dcerpc_sec_vt_pcontext pcontext = {
 		.abstract_syntax = pipe_fns->syntax,
 		.transfer_syntax = ndr_transfer_syntax,
@@ -1573,7 +1572,7 @@ static bool srv_pipe_check_verification_trailer(struct pipes_struct *p,
 		goto done;
 	}
 
-	ret = dcerpc_sec_verification_trailer_check(vt, &bitmask1,
+	ret = dcerpc_sec_verification_trailer_check(vt, NULL,
 						    &pcontext, &header2);
 done:
 	TALLOC_FREE(frame);
-- 
1.9.1

